### PR TITLE
feat(GBB-28): get users api and pagination

### PR DESCRIPTION
**Pull request by LeTatTuan**
@phamhongphuc1403 @duchuyyyy @ThuongHoai081

### CODE REVIEW

File: src/controller/user.controller.js [hunk 1]
```diff
@@ -23,5 +24,19 @@ class userController {
...
+    const { sort } = req.query;
+    let sortCriteria;
+    if (sort) {
+      sortCriteria = sort.split(',').map((item) => {
+        const [field, order] = item.split(':');
+        return [field, order];
+      });
+    }
+    new SuccessResponse({
+      data: await UserService.getUsers(PaginationDto({ ...req.query, sort: sortCriteria })),
```
Comment (High10Hunter) [inline src/controller/user.controller.js]: @LeTatTuan should split the`PaginationDto({ ...req.query, sort: sortCriteria }` part as a separate variable for better readability
e.g: `paginationDto = PaginationDto({ ...req.query, sort: sortCriteria }`

File: src/validations/user.validation.js [hunk 2]
```diff
@@ -20,3 +20,12 @@ export const updateUserByAdmin = {
     role_id: JoiUtils.id(),
   }),
 };
+
+export const getUsers = {
```
Comment (High10Hunter) [inline src/validations/user.validation.js]: @LeTatTuan should rename this variable as `getUsersPagination`

**Review by High10Hunter — CHANGES_REQUESTED** [review]

File: src/controller/user.controller.js (hunk 1 above)
Comment (LeTatTuan) [reply src/controller/user.controller.js]: @High10Hunter used variable for more clarity

File: src/validations/user.validation.js (hunk 2 above)
Comment (LeTatTuan) [reply src/validations/user.validation.js]: changed name to getUsersPagination

**Pull request merged**